lib/board.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Board do
+  @moduledoc """
+  The module that handles the Board logic for a Tic-Tac-Toe game
+  """
+
+  @spec setup_initial_board(non_neg_integer()) :: map()
+  def setup_initial_board(board_dimensions \\ 3) do
+    Map.new(1..board_dimensions * board_dimensions, fn position -> {position, "#{position}"} end)
+  end
+
+  @spec update_board(map(), String.t(), String.t()) :: map()
+  def update_board(unformatted_board, player_move, symbol) do
```

Comment:
Nitpick: is there going to be a `formatted_board` or is it ok if we just call it `board`?